sunpy/instr/rhessi.py

```diff
@@ -219,7 +219,55 @@ def get_obssum_filename(time_range):
 
     filenames = dbase_dat.get('filename')[index_number_start:index_number_end]
     return [posixpath.join(get_base_url(), 'metadata', 'catalog', filename + 's')
-            for filename in filenames]
+        for filename in filenames]
+
+
+def get_obssum_filename_multiple_months(time_range):
+    """
+    Download the RHESSI observing summary data from one of the RHESSI
+    servers, parses it, and returns the name of the obssumm files relevant for
+    the time range.
+
+    Parameters
+    ----------
+    time_range : str, TimeRange
+        A TimeRange or time range compatible string
+
+    Returns
+    -------
+    out : list of lists
+        Returns the filenames of the observation summary file for multiple months
+        in the time range
+
+    Examples
+    --------
+    >>> import sunpy.instr.rhessi as rhessi
+    >>> rhessi.get_obssum_filename_multiple_months(('2011/04/04', '2011/04/05'))   # doctest: +SKIP
+    ['http://soleil.i4ds.ch/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits']
+
+    .. note::
+        This API is currently limited to providing data from whole days only.
+
+    """
+    time_range = TimeRange(time_range)
+    filenames = []
+    delta = relativedelta(time_range.end, time_range.start)
+    date = time_range.start
+    if  delta.years <= 0 and delta.months <= 0 and time_range.start.month != time_range.end.month :
```

Comment:
Is this code here just to check that the `time_range` here is less than one month and which case the user should not be using this function at all, correct?

Reply:
Its partially true , Actually `get_obssum_filename` cannot return filenames spanning over multiple months even its less than 1 month. (Ex . `Time_range(("2011/04/28"),("2011/05/05"))` ). Rhessi servers return data of a single month when `get_obssum_filename` , is called ,so this code makes sure that if `time_range` is less than a month and spans over 2 month then ``get_obssum_filename` should be called 2 times .